packages/grid/_modules_/grid/components/column-header-item.tsx

```diff
@@ -44,7 +42,21 @@ export const ColumnHeaderItem = React.memo(
       });
     }
 
-    const handleResize = onResizeColumn ? () => onResizeColumn(column) : undefined;
+    const handleResize =
+      onResizeColumn && !disableColumnReorder ? () => onResizeColumn(column) : undefined;
```

Comment:
by doing ` () => onResizeColumn(column) ` you create a new function on every render.
This is going to break the optimisation of using react memo when you pass the function to the subcomponents
as the shallow equal will just be false.
You can fix this by using useCallback

Reply:
There is a memo in the part, this is a leaf component, it doesn't matter

Reply:
I have handled this aspect in #436. The handler is moved to the child element, so the memo isn't always a noop.

Reply:
Actually, I don't know why I added this here but I tested it without `!disableColumnReorder` and it works. I've reverted it to its original state.